src/main.rs

```diff
@@ -287,6 +341,57 @@ async fn main() {
         ),
     };
 
+    let mut validator_config = ValidatorConfig {
+        tpu_enable_udp: matches.is_present("tpu_enable_udp"),
+        tpu_disable_quic: matches.is_present("tpu_disable_quic"),
+        max_ledger_size: if matches.is_present("limit_ledger_size") {
+            let limit_ledger_size = match matches.value_of("limit_ledger_size") {
+                Some(_) => value_t_or_exit!(matches, "limit_ledger_size", u64),
+                None => DEFAULT_MAX_LEDGER_SHREDS,
+            };
+            if limit_ledger_size < DEFAULT_MIN_MAX_LEDGER_SHREDS {
+                error!(
+                    "The provided --limit-ledger-size value was too small, the minimum value is {DEFAULT_MIN_MAX_LEDGER_SHREDS}"
+                );
+                return;
```

Comment:
nit: I just found out about this, but a nice option if there's an issue on arg parsing that can't be covered by clap https://github.com/solana-labs/solana-program-library/pull/6550#discussion_r1580158113

Doesn't need to be done here, of course